core/src/main/java/io/grpc/ClientCallImpl.java

```diff
@@ -127,70 +145,148 @@ public void start(Listener<RespT> observer, Metadata.Headers headers) {
       headers.put(HttpUtil.USER_AGENT_KEY, userAgent);
     }
 
+    boolean started = startInternal(listener, headers);
+    // Start the deadline timer after stream creation because it will close the stream
+    if (started && deadlineNanoTime != null) {
+      deadlineCancellationFuture = startDeadlineTimer(timeoutMicros);
+    }
+  }
+
+  private boolean startInternal(
+      final ClientStreamListener listener, final Metadata.Headers headers) {
+    ClientTransport transport;
     try {
-      stream = transport.newStream(method, headers, listener);
-    } catch (IllegalStateException ex) {
-      // We can race with the transport and end up trying to use a terminated transport.
-      // TODO(ejona86): Improve the API to remove the possibility of the race.
+      transport = clientTransportProvider.get();
+    } catch (RuntimeException ex) {
       closeCallPrematurely(listener, Status.fromThrowable(ex));
+      return false;
     }
-    // Start the deadline timer after stream creation because it will close the stream
-    if (deadlineNanoTime != null) {
-      deadlineCancellationFuture = startDeadlineTimer(timeoutMicros);
+    if (transport == null) {
+      closeCallPrematurely(listener, Status.UNAVAILABLE.withDescription("Channel is shutdown"));
+      return false;
     }
+
+    synchronized (lock) {
+      try {
+        stream = transport.newStream(method, headers, listener);
+        // This will typically be empty, and no calls will be added to it once stream is not null.
+        for (Runnable queuedCall : queuedOperations) {
+          queuedCall.run();
+        }
+      } catch (IllegalStateException ex) {
+        // We can race with the transport and end up trying to use a terminated transport.
+        // TODO(ejona86): Improve the API to remove the possibility of the race.
+        retryFuture = scheduledExecutor.schedule(new Runnable() {
+          @Override
+          public void run() {
+            startInternal(listener, headers);
```

Comment:
Done.  There is a tricky part to this where startInternal could have an unexpected exception and not reschedule itself.  As long at the rpc specified a deadline I believe it will eventually finish.  Otherwise this call will just be dead.

Also, moved the queuedOperations outside of the catch to prevent unexpected ISE's from trigger a retry.  (Exceptions as flow control, yikes).